cache/cache.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Cache do
+  use GenServer
+
+  @type result ::
+          {:ok, any()}
+          | {:error, :timeout}
+          | {:error, :not_registered}
+
+  @impl GenServer
+  def start_link(opts \\ []) do
+  end
+
+  @doc ~s"""
+  Registers a function that will be computed periodically to update the cache.
+
+  Arguments:
+    - `fun`: a 0-arity function that computes the value and returns either
+      `{:ok, value}` or `{:error, reason}`.
+    - `key`: a term which is associated with the function and is used to
```

Comment:
We can make the key a string for simplicity, but can be just a term too.

Reply:
It shouldn't really matter at all TBH.

Reply:
I've adjusted the wording accordingly 👍